server/apps/client.go

```diff
@@ -7,41 +7,42 @@ import (
 	"encoding/json"
 	"io"
 	"net/http"
-	"path"
 	"time"
 
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+
 	"github.com/dgrijalva/jwt-go"
+	"github.com/mattermost/mattermost-plugin-apps/server/store"
 )
 
 type Client interface {
-	PostWish(Call) (*CallResponse, error)
-	PostChangeNotification(Subscription, interface{})
-	GetManifest(manifestURL string) (*Manifest, error)
+	PostWish(*Call) (*CallResponse, error)
+	PostNotification(*NotificationRequest) error
+	GetManifest(manifestURL string) (*store.Manifest, error)
 }
 
 const OutgoingAuthHeader = "Mattermost-App-Authorization"
 
-func (s *Service) PostChangeNotification(sub Subscription, msg interface{}) {
-	app, err := s.Registry.Get(sub.AppID)
+func (s *service) PostNotification(n *NotificationRequest) error {
+	app, err := s.Store.GetApp(n.Context.AppID)
 	if err != nil {
-		// <><> TODO log
-		return
+		return err
 	}
 
-	resp, err := s.post(app, "", path.Join(app.Manifest.RootURL, "notify", string(sub.Subject)), msg)
+	resp, err := s.post(app, "", app.Manifest.RootURL+"/notify/"+string(n.Subject), n)
```

Comment:
Nit: Why not use the "path.Join"? I would even prefer a `fmt.Sprintf` than just the `+`. That way probably the string cast can be removed.